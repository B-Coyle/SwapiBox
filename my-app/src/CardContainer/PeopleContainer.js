import React, {Component} from 'react';
import {fetchPeople, fetchHomeworld, fetchSpecies} from '../ApiCall/apiCalls.js';
import Card from '../Card/Card.js';
import './Container.scss';

const uuidv4 = require("uuid/v4");

export default class PeopleContainer extends Component{
    constructor() {
        super()
        this.state= {
            people: [],
            error: '',
            loading: false
        }
    }

    componentDidMount() {
        this.setState({
          loading: true
        });

        this.getPeople();
      }
    
      getPeople = () => {
        fetchPeople()
          .then(data => fetchHomeworld(data.results))
          .then(people => fetchSpecies(people))
          .then(people => this.formatPerson(people))
          .catch(error => this.setState({ error }));
      };
    
      formatPerson = people => {
        let updatedPeople = people.map(person => {
          let { name, homeworld, population, species } = person;
          return { name, homeworld, population, species, key: uuidv4() };
        });

        this.setState({ 
          people: updatedPeople, 
          loading: false 
        });
        
      }; 

      render() {
        const peopleCards = this.state.people.map(person => 
          <Card 
            key={person.key}
            data={person}
            category="person"
          />
        )
    
        const loadingMessage = (
          <div className="loadingMessage">
            <p>Information loading. Please Hold...</p>
          </div>
        );
        return (
          <section className="cardContainer">
            {this.state.loading ? loadingMessage : peopleCards}
          </section>
        );
      }
    }

