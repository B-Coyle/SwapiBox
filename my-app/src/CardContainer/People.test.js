import React from 'react';
import PeopleContainer from '../CardContainer/PeopleContainer.js';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';


describe('PeopleContainer', () => {
    let wrapper; 
    const defaultState = {
        people: [],
        error: "",
        loading: true
    };
    
      beforeEach(()=> {
        wrapper = shallow(<PeopleContainer/>);

      });

    it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
    });

    it("should have default state", () => {
    expect(wrapper.state()).toEqual(defaultState);
    });

})