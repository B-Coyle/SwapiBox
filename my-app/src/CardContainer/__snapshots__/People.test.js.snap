// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleContainer should match the snapshot with all data passed in 1`] = `
<section
  className="cardContainer"
>
  <div
    className="loadingMessage"
  >
    <p>
      Information loading. Please Hold...
    </p>
  </div>
</section>
`;
